pkg/kube/inject/inject.go

```diff
@@ -780,7 +782,7 @@ func IntoObject(sidecarTemplate string, valuesConfig string, revision string, me
 	// due to bug https://github.com/kubernetes/kubernetes/issues/57923,
 	// k8s sa jwt token volume mount file is only accessible to root user, not istio-proxy(the user that istio proxy runs as).
 	// workaround by https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod
-	if meshconfig.SdsUdsPath != "" {
+	if constants.DefaultSdsUdsPath != "" {
```

Comment:
Do we need this check?

Reply:
We don't need this check. I tried to cleanup this codepath in a different PR but it was blocked by different bootstrap config for TestEnvoy which didn't have sds cluster

Reply:
Oh sorry, this is a different location- yeah we don't need this check I will remove it